# CINSignatureQuantification

<!-- badges: start -->
[![minimal R version](https://img.shields.io/badge/R%3E%3D-3.6.0-blue.svg)](https://cran.r-project.org/)
<!-- badges: end -->

R package to quantify signatures of chromosomal instability on absolute copy number profiles as described in [Drews et al. (Nature, 2022)](https://www.nature.com/articles/s41586-022-04789-9).

# Introduction

Chromosomal instability (CIN) results in the accumulation of large-scale losses, gains, and rearrangements of DNA. In our recent study [1], we present a systematic framework to measure different types of CIN and their impact on clinical phenotypes. This R package allows you to quantify the activity of the 17 signatures presented. It also allows you to quantify signature activities from other publications [2]. First, copy number features are extracted from the copy number profiles. Second, the features are assigned to components for which a probability will be calculated. These probabilities are then summed up for each patient. Third, the probabilities across the components are used to quantify signature activities. Fourth, the signature activities can be used to predict patient response to platinum-based chemotherapies.

The `CINSignatureQuantification` package provides you with the functions and example data to automise this process of quantifying copy number signature activities.

# Installation

Install the package from Github directly by using `devtools`
```r
devtools::install_github("markowetzlab/CINSignatureQuantification", build_vignettes = TRUE, dependencies = TRUE)
```

Then load the package with:
```r
library(CINSignatureQuantification)
```
# Input data

Input data is absolute copy number profiles in segment table format, containing multiple samples, without allele or subclonal information, see below. 

## Example input

|chromosome |start     |end         |segVal    |sample      |
|-----------|----------|------------|----------|------------|
|1          |61735     |249224388   |3.1       |TCGA-BT-A20P|
|2          |12784     |82571206    |2.0       |TCGA-BT-A20P|
|3          |82571664  |85357333    |0.843     |TCGA-BT-A20P|

Input data processing can have an impact on ability to robustly quantify copy number signatures across differing technologies. We describe best practises to limit these differences in the [supplemental methods (Drews et al. 2022)](https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-022-04789-9/MediaObjects/41586_2022_4789_MOESM1_ESM.pdf) but the general recommendations are as follows:

- SNP6 arrays: Recommended use with and without matched normals using ASCAT
with a penalty of 70.
- WGS: Recommended use with ASCAT focussing on SNP6 positions or ABSOLUTE
on full WGS profiles.
- Shallow WGS: Recommended use with ASCAT.sc and a penalty of 0.001
- WES: For on-target reads, either ASCAT (penalty 70) or Sequenca can be used. For
off-target reads, ASCAT.sc with penalty 0.001 is recommended.

**Warning:** It is preferable use _unrounded_ copy number segments (floating point segVal states). _Rounded_ copy number states (integer segVal states) will work but outputs may not be directly comparable to _unrounded_ copy number outputs.

## Example data

The package comes with a set of 478 samples that were both part of the TCGA and the PCAWG cohort and have detectable levels of CIN [1] contained within the github repository.

```
data("TCGA_478_Samples_SNP6_GOLD")
```

# Quick start

The `CINSignatureQuantification` package offers two main functions: `quantifyCNSignatures` and `clinPredictionPlatinum`. It also allows you to do the signature quantification step-by-step with these functions: `createCNQuant`, `calculateFeatures`, `calculateSampleByComponentMatrix`, `calculateActivity` and `clinPredictionDenovo`.

## Quantifying signature activies

CIN signatures can be quantified using the `quantifyCNSignatures()` function where the primary input is a a segment table.

```r
mySigs = quantifyCNSignatures(TCGA_478_Samples_SNP6_GOLD)
```

Input can be a loaded R data.frame object or file path to a segment table. There is also support to load segment tables directly from a `QDNAseqCopyNumbers` object generated by [QDNAseq](https://github.com/ccagc/QDNAseq). By default, the build used is `hg19` and method is `drews`, see function documentation for additional options.

# Maintenance & help

For any code bugs, feature requests, or implementation errors, please open an issue. For more information on obtaining copy number profiles, please refer to the documentation of common copy number callers like [ASCAT](https://github.com/VanLoo-lab/ascat) or [ABSOLUTE](https://github.com/ShixiangWang/DoAbsolute). More information on how to work with and generate copy number signatures can be obtained from: [Drews et al. (Nature, 2022)](https://www.nature.com/articles/s41586-022-04789-9) or [Macintyre et al. (Nature Genetics, 2018)](https://www.nature.com/articles/s41588-018-0179-8).

# Citation
Please cite `CINSignatureQuantification` as described:

```
citation("CINSignatureQuantification")
```
# References

* [1] [Drews et al. (Nature, 2022)](https://www.nature.com/articles/s41586-022-04789-9)
* [2] [Macintyre et al. (Nature Genetics, 2018)](https://www.nature.com/articles/s41588-018-0179-8)

# Authors

[Ruben Drews](https://github.com/Martingales) Ruben.Drews 'at' cruk.cam.ac.uk

[Philip Smith](https://github.com/Phil9S) Philip.Smith 'at' cruk.cam.ac.uk


# Licence
The contents of this repository are copyright (c) 2022, University of Cambridge and Spanish National Cancer Research Centre (CNIO).

The contents of this repository are published and distributed under the GAP Available Source License v1.0 (ASL). 

The contents of this repository are distributed in the hope that it will be useful for non-commercial academic research, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the ASL for more details. 

The methods implemented in the code are the subject of pending patent application GB 2114203.9.

Any commercial use of this code is prohibited.
